starfish/imagestack/imagestack.py

```diff
@@ -711,18 +714,53 @@ def transform(
             group_by = {Indices.X, Indices.Y}
 
         indices = list(self._iter_indices(group_by))
+        slice_list = [self._build_slice_list(index)[0]
+                      for index in indices]
 
+        indices_and_slice_list = zip(indices, slice_list)
         if verbose:
-            tiles = tqdm(self._iter_tiles(indices))
-        else:
-            tiles = self._iter_tiles(indices)
+            indices_and_slice_list = tqdm(indices_and_slice_list)
 
-        applyfunc: Callable = partial(func, **kwargs)
+        applyfunc: Callable = partial(
+            self._multiprocessing_workflow,
+            _StripArguments(partial(func, **kwargs), (1, 2)),
+            finalizer_callable,
+        )
 
-        with multiprocessing.Pool(n_processes) as pool:
-            results = pool.imap(applyfunc, tiles)
+        with multiprocessing.Pool(
+                n_processes,
+                initializer=shmem.initializer,
+                initargs=((self._data._backing_mp_array,
+                           self._data._data.shape,
+                           self._data._data.dtype),)) as pool:
+            results = pool.imap(applyfunc, indices_and_slice_list)
             return list(zip(results, indices))
 
+    @staticmethod
+    def _multiprocessing_workflow(
+            worker_callable: Callable[[np.ndarray,
+                                       Mapping[Indices, int],
+                                       Tuple[Union[int, slice], ...]], Any],
+            finalizer_callable: Optional[
+                Callable[[np.ndarray,
+                          Any,
+                          Mapping[Indices, int],
+                          Tuple[Union[int, slice], ...]], Any]],
+            indices_and_slice_list: Tuple[Mapping[Indices, int],
+                                          Tuple[Union[int, slice], ...]],
+    ):
+        backing_mp_array, shape, dtype = shmem.get_payload()
+        unshaped_numpy_array = np.frombuffer(backing_mp_array.get_obj(), dtype=dtype)
+        numpy_array = unshaped_numpy_array.reshape(shape)
+
+        sliced = numpy_array[indices_and_slice_list[1]]
```

Comment:
So, we're doing something to `sliced`, and `finalizer_callable` is enabling us to mess around with the shared numpy array (e.g., enables in-place operations?)

Wouldn't that technically also be available within worker_callable() if it did any in-place operation?

Reply:
the sliced data is available in `worker_callable`.  the entire original data set is not available.  i guess we could conceivably limit the view to the slice as far as the finalizer is concerned.  i was trying to generalize it, but I can definitely see that it may be premature generalization.